src/rebar3_hex_publish.erl

```diff
@@ -69,21 +66,45 @@ do(State) ->
             ?PRV_ERROR(Reason)
     end.
 
+%% Revert cases 
 handle_task(#{args := #{revert := undefined}}) ->
     {error, "--revert requires an app version"};
 
-handle_task(#{args := #{revert := Vsn}, repo := Repo, state := State}) ->
+handle_task(#{args := #{revert := Vsn}, multi_project := false} = Task) ->
+    #{repo := Repo, state := State} = Task,
     App = rebar_state:current_app(State),
     Name = rebar_app_info:name(App),
     ok = rebar3_hex_revert:revert(binarify(Name), binarify(Vsn), Repo, State),
     {ok, State};
 
-handle_task(#{repo := Repo, state := State}) ->
+handle_task(#{args := #{revert := Vsn, app := AppName}, multi_project := true} = Task) ->
```

Comment:
This is in a sub-optimal state still, but I wanted to unblock @paulo-ferraz-oliveira and get this in real quick vs refactoring right now.